Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/BookingCancellationReason.hs

```diff
@@ -25,6 +26,22 @@ import Storage.Tabular.BookingCancellationReason
 create :: BookingCancellationReason -> SqlDB ()
 create = Esq.create
 
+findAllCancelledByDriverId ::
+  Transactionable m =>
+  Id Person ->
+  m Int
+findAllCancelledByDriverId driverId = do
+  mkCount <$> do
+    Esq.findAll $ do
+      rideBookingCancellationReason <- from $ table @BookingCancellationReasonT
```

Comment:
why finding in cacelation reason table? You should find in booking table?

Reply:
Booking table doesn't have driver id will have to look in ride table , and to penalise a driver we will eventually have to fetch cancelled by driver from this table.